src/Settings.js

```diff
@@ -162,6 +169,41 @@ export class Settings {
         }
     }
 
+    /**
+     * @returns {SiteSpecificFeature|undefined}
+     */
+    get siteSpecificFeature() {
+        return this._siteSpecificFeature ?? undefined;
+    }
+
+    async getsiteSpecificFeature() {
```

Comment:
This looks like the correct place to add this - can you include some sample JSON examples in integration tests? the tests are setup to allow you to pass this contentScope stuff, so I'd focus on getting some HTML examples with corresponding JSON config (maybe in a new folder/format?)

This will also help with code review too, since it will highlight the design of the remotely delivered data